src/commonMain/kotlin/exchange.dydx.abacus/calculator/AccountCalculator.kt

```diff
@@ -19,17 +19,100 @@ class AccountCalculator(val parser: ParserProtocol) {
     ): Map<String, Any>? {
         return if (account != null) {
             val subaccounts = parser.asMap(account["subaccounts"]) ?: return account
-            val modified = account.mutable()
+            var modified = account.mutable()
             for ((subaccountNumber, subaccount) in subaccounts) {
                 val parentSubaccountNumber = subaccountNumber.toInt() % kChildSubaccountMod
                 if (parentSubaccountNumber in subaccountNumbers) {
                     val key = "subaccounts.$subaccountNumber"
                     modified.safeSet(key, subaccountCalculator.calculate(parser.asNativeMap(subaccount), configs, markets, price, periods))
                 }
             }
+            modified = groupSubaccounts(modified)
             modified
         } else {
             null
         }
     }
+
+    private fun groupSubaccounts(existing: Map<String, Any>): MutableMap<String, Any> {
+        val modified = existing.mutable()
+        val subaccounts = parser.asNativeMap(parser.value(existing, "subaccounts"))
+        val subaccountNumbers = subaccounts?.keys?.mapNotNull { parser.asInt(it) }?.sorted()
+        if (subaccountNumbers != null) {
+            val groupedSubaccounts = mutableMapOf<String, Any>()
+            for (subaccountNumber in subaccountNumbers) {
+                val subaccount =
+                    parser.asNativeMap(parser.value(subaccounts, "$subaccountNumber")) ?: break
+                if (subaccountNumber < 128) {
```

Comment:
Utilize the constant instead of using magic number. Might be worth adding to `utils/constants` as `NUM_PARENT_SUBACCOUNT` then it can also be utilized in `AccountSupervisors` `kChildSubaccountMod`